ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/HD/DiffSeq.hs

```diff
@@ -0,0 +1,354 @@
+{-# LANGUAGE ConstraintKinds            #-}
+{-# LANGUAGE DataKinds                  #-}
+{-# LANGUAGE DeriveAnyClass             #-}
+{-# LANGUAGE DeriveFunctor              #-}
+{-# LANGUAGE DeriveGeneric              #-}
+{-# LANGUAGE DerivingVia                #-}
+{-# LANGUAGE FlexibleContexts           #-}
+{-# LANGUAGE FlexibleInstances          #-}
+{-# LANGUAGE GeneralisedNewtypeDeriving #-}
+{-# LANGUAGE MultiParamTypeClasses      #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+{-# LANGUAGE TypeFamilies               #-}
+
+{- | Sequences of diffs for ledger tables.
+
+   These diff sequences are an instantiation of a strict finger tree with root
+   measures. The tree/sequence itself contains diffs and slot information, while
```

Comment:
I don't know if we should refer to `slots` in this level of abstraction. What about something like "sequence numbers"?

Reply:
The specific datatype does contain slot information, so I'm not sure if we can replace it -- Or do you mean to say that the diff sequence should abstract over different kinds of "sequence numbers", like slots?